app/Insightly/SyncIsAllowed.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Insightly;
+
+use App\Domain\Contacts\Contact;
+use App\Domain\Contacts\ContactType;
+
+final class SyncIsAllowed
+{
+    public static function forContact(Contact $contact): bool
+    {
+        return match ($contact->type) {
+            ContactType::Functional, ContactType::Technical => true,
+            ContactType::Contributor => false,
+        };
```

Comment:
Very small nit: Since the return value is a boolean, you could also do it without a `match` like:
```
return $contact->type === ContactType::Functional || $contact->type === ContactType::Technical;
```

Or even:
```
return in_array($contact->type, [ContactType::Functional, ContactType::Technical]);

Reply:
I prefer the match statement as I find it more readable and works incredibly well with enums. An IDE can check whether you have matched all possible enum cases, which might also be added to a static analysis tool. This will detect potential issues when you add a new enum case, for instance.